packages/amplify-migration-e2e/src/__tests__/migration_templategen_e2e.test.ts

```diff
@@ -0,0 +1,50 @@
+import path from 'node:path';
+import assert from 'node:assert';
+import { createNewProjectDir, npmInstall, deleteS3Bucket } from '@aws-amplify/amplify-e2e-core';
+import { assertDefaultGen1Setup } from '../assertions';
+import { setupAndPushDefaultGen1Project, runCodegenCommand, runGen2SandboxCommand, cleanupProjects } from '..';
+import { copyFunctionFile } from '../function_utils';
+import { copyGen1Schema } from '../api_utils';
+import { updatePackageDependency } from '../updatePackageJson';
+import { createS3Bucket } from '../sdk_calls';
+import { runTemplategenCommand, stackRefactor } from '../templategen';
+
+void describe('Templategen E2E tests', () => {
+  void describe('Full Migration Templategen Flow', () => {
+    let projRoot: string;
+    let projName: string;
+    let bucketName: string;
+
+    beforeEach(async () => {
+      const baseDir = process.env.INIT_CWD ?? process.cwd();
+      projRoot = await createNewProjectDir('templategen_e2e_flow_test', path.join(baseDir, '..', '..'));
+      projName = `test${Math.floor(Math.random() * 1000000)}`;
+      bucketName = `testbucket${Math.floor(Math.random() * 1000000)}`;
```

Comment:
we have `generateRandomShortId` somewhere in `amplify-e2e-core` for this kind of thing.

Reply:
Thanks for pointing this out! [b28e101](https://github.com/aws-amplify/amplify-cli/pull/13993/commits/b28e1014fa504a64624b444c916fde099102071e)